### PR TITLE
[Document]: Error/undefined params to the 'done' callback

### PR DESCRIPTION
**Pull request by maraisr**
Resolves: #3134

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -155,7 +155,7 @@ Mocha allows you to use any assertion library you wish.  In the above example, w
 
 ## Asynchronous Code
 
-Testing asynchronous code with Mocha could not be simpler! Simply invoke the callback when your test is complete. By adding a callback (usually named `done`) to `it()`, Mocha will know that it should wait for this function to be called to complete the test.
+Testing asynchronous code with Mocha could not be simpler! Simply invoke the callback when your test is complete. By adding a callback (usually named `done`) to `it()`, Mocha will know that it should wait for this function to be called to complete the test. This callback accepts both an `Error` instance, or `undefined`.
```
Comment (boneskull) [inline docs/index.md]: #3134 isn't quite accurate about what needs to be documented.

We need to document that *any non-falsy, non-`Error` instance* passed into `done()` will cause Mocha to fail.

Subclasses of `Error` are allowed.

File: docs/index.md [hunk 2]
```diff
@@ -184,6 +184,7 @@ describe('User', function() {
 });
 ```
 
+
```
Comment (boneskull) [inline docs/index.md]: please remove

File: docs/index.md (hunk 1 above)
Comment (maraisr) [reply docs/index.md]: Can you check if that is accurate?

**Comment by maraisr** [comment]
@boneskull do you think we can get this merged? Kinda excited to have committed something that was merged into Mocha.

File: docs/index.md [hunk 3]
```diff
@@ -171,7 +171,7 @@ describe('User', function() {
 });
 ```
 
-To make things even easier, the `done()` callback accepts an error, so we may use this directly:
+To make things even easier, the `done()` callback also accepts an `Error` instance (ie, `new Error()`), so we may use this directly:
```
Comment (boneskull) [inline docs/index.md]: Should be `i.e.` not `ie`

File: docs/index.md [hunk 4]
```diff
@@ -155,7 +155,7 @@ Mocha allows you to use any assertion library you wish.  In the above example, w
 
 ## Asynchronous Code
 
-Testing asynchronous code with Mocha could not be simpler! Simply invoke the callback when your test is complete. By adding a callback (usually named `done`) to `it()`, Mocha will know that it should wait for this function to be called to complete the test.
+Testing asynchronous code with Mocha could not be simpler! Simply invoke the callback when your test is complete. By adding a callback (usually named `done`) to `it()`, Mocha will know that it should wait for this function to be called to complete the test. This callback accepts both an `Error` (or subclassed) instance, or `undefined`, any non-falsy will cause a failed test.
```
Comment (boneskull) [inline docs/index.md]: Sorry, let's just eliminate `undefined` altogether here.  it's also a bit of a run-on sentence.  how about:

```markdown
This callback accepts both an `Error` instance (or subclass thereof) *or* a falsy value; anything else will cause a failed test.
```

**Comment by boneskull** [comment]
thanks!

**Pull request merged**